Sources/ViewControllers/Products/Search/SearchTableViewController.swift

```diff
@@ -73,19 +73,19 @@ class SearchTableViewController: UITableViewController, DataManagerClient {
     }
 
     fileprivate func updateInitialView() {
-        initialView.loadingProgressView.setProgress(Float(offlineStatus.percent/100), animated: true)
-        initialView.loadingTitleLabel.text = String.localizedStringWithFormat(NSLocalizedString("product-search.initial-view.offline.title-loading", comment: ""), Int(offlineStatus.percent))
-        initialView.loadingSubtitleLabel.text = String.localizedStringWithFormat(NSLocalizedString("product-search.initial-view.offline.subtitle", comment: ""), offlineStatus.savedProductsCount)
+        initialView.loadingProgressView?.setProgress(Float(offlineStatus.percent/100), animated: true)
```

Comment:
While this would probably work by making the code not crash, we should try to find why this view is nil at the moment. From the stack trace it seems that it happens in a closure inside viewDidLoad, so when the offlineStatus change. There is something weird here, in the fact that the observeToken  is stored but never used to stop listening. My guess is that the viewDidLoad disappear at some time, but the offline listener still try to trigger an update, which means the view is indeed nil. The proper fix in this case is to call `observeNotificationToken?.invalidate()` in `viewWillDisappear`, and move `self.observeNotificationToken = self.offlineStatus.observe { (_) in
            self.updateInitialView()
        }` in viewWillAppear.

Reply:
I agree, it is not a real solution.